pyomo/contrib/incidence_analysis/interface.py

```diff
@@ -866,35 +884,72 @@ def remove_nodes(self, nodes, constraints=None):
 
         Parameters
         ----------
-        nodes: list
-            VarData or ConData objects whose columns or rows will be
-            removed from the incidence matrix.
+        variables: list
+            VarData objects whose nodes will be removed from the incidence graph
         constraints: list
-            VarData or ConData objects whose columns or rows will be
-            removed from the incidence matrix.
+            ConData objects whose nodes will be removed from the incidence graph
+
+        .. note::
+
+           **Deprecation in Pyomo vTBD**
+
+           The pre-TBD implementation of ``remove_nodes`` allowed variables and
+           constraints to remove to be specified in a single list. This made
+           error checking difficult, and indeed, if invalid components were
+           provided, we carried on silently instead of throwing an error or
+           warning. As part of a fix to raise an error if an invalid component
+           (one that is not part of the incidence graph) is provided, we now require
+           variables and constraints to be specified separately.
 
         """
         if constraints is None:
             constraints = []
+        if variables is None:
+            variables = []
         if self._incidence_graph is None:
             raise RuntimeError(
                 "Attempting to remove variables and constraints from cached "
                 "incidence matrix,\nbut no incidence matrix has been cached."
             )
-        to_exclude = ComponentSet(nodes)
-        to_exclude.update(constraints)
-        vars_to_include = [v for v in self.variables if v not in to_exclude]
-        cons_to_include = [c for c in self.constraints if c not in to_exclude]
+
+        vars_to_validate = []
+        cons_to_validate = []
+        depr_msg = (
+            "In IncidenceGraphInterface.remove_nodes, passing variables and"
+            " constraints in the same list is deprecated. Please separate your"
+            " variables and constraints and pass them in the order variables,"
+            " constraints."
+        )
```

Comment:
The logic below will emit this message for each variable in the list.  While the `deprecation_warning` logic should suppress the duplicate messages, it would probably be better to not emit them in the first place.  Would it be simpler to just run through the loops below and then
```python
for var in variables:
    if var in self._var_index_map:
        vars_to_validate.append(var)
    elif var in sef._con_index_map:
        cons_to_validate.append(var)
    else:
        unknown_comp.append(var)
if cons_to_validate:
    deprecation_warning(depr_msg, version="6.7.2.dev0")
nVars = len(vars_to_validate)
for var in variables:
    if var in self._var_index_map:
        vars_to_validate.append(var)
    elif var in sef._con_index_map:
        cons_to_validate.append(var)
    else:
        unknown_comp.append(var)
if len(vars_to_validate) != nVars:
    deprecation_warning(depr_msg, version="6.7.2.dev0")
if unknown_comp:
    raise ValueError(...)
```